### PR TITLE
Add event generics so that the target element can be customized.

### PR DESCRIPTION
**Pull request by milesj**
Closes https://github.com/adobe/react-spectrum/issues/1526

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

None really since it's just types.

### CODE REVIEW

File: packages/@react-types/shared/src/events.d.ts [hunk 1]
```diff
@@ -12,29 +12,27 @@
...
 // Event bubbling can be problematic in real-world applications, so the default for React Spectrum components
 // is not to propagate. This can be overridden by calling continuePropagation() on the event.
-export type BaseEvent<T extends SyntheticEvent> = T & {
+export type BaseEvent<T extends SyntheticEvent<HTMLElement>> = T & {
   /** @deprecated Use continuePropagation. */
   stopPropagation(): void,
   continuePropagation(): void
 }
 
-export type KeyboardEvent = BaseEvent<ReactKeyboardEvent<any>>;
```
Comment (milesj) [inline packages/@react-types/shared/src/events.d.ts:27]: Not sure why this existed, but removing it fixed some issues.

**Comment by milesj** [comment]
I'm having some trouble getting the docs to build locally. All my console logs and debugging are not working :/

File: packages/@react-types/shared/src/events.d.ts (hunk 1 above)
Comment (snowystinger) [reply packages/@react-types/shared/src/events.d.ts:27]: I don't think we can remove this export, it's in a published package.
I imagine it's also the reason for your docs build failing https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/interactions/src/usePress.ts#L273

**Review by snowystinger — COMMENTED** [review]
Thanks for the PR, this is a complex change, we'll need some time to review this.
It'd be helpful if you could supply an application where we could reproduce the errors you're seeing, some project where we can clone it and immediately try it and know it's the same as the concerns you have.

**Comment by Kerumen** [comment]
@milesj @snowystinger Can I help this PR move forward? I also need this in my current project.

**Comment by milesj** [comment]
@Kerumen We actually went a different direction, so no longer need this. You're welcome to copy/take this over 😅

**Comment by Kerumen** [comment]
@milesj Can you point me your solution? Thanks!

**Comment by milesj** [comment]
@Kerumen We stopped using react-aria 😅

**Comment by milesj** [comment]
With that being said, gonna close this.

**Pull request closed without merging**